### ISSUE TITLE
Keychain file names changed on macOS 10.12 (beta)

### ISSUE THREAD

**Issue by dgoodlad** [99designs/aws-vault#84]
The names given to keychains in `~/Library/Keychains` have changed to have the extension `.keychain-db` rather than `.keychain` on beta versions of macOS 10.12.

While running `aws-vault add` for the first time works correctly, subsequent operations (e.g. `aws-vault exec`) will not find credentials, and will never even request access to the keychain file. The file that was created is `~/Library/Keychains/aws-vault.keychain-db`, despite debug output during `aws-vault add` saying it was opening `~/Library/Keychains/aws-vault.keychain`.

I had a poke around and modified [keychain.go:44](https://github.com/99designs/aws-vault/blob/42854c4d7d2af53a0978244ee20db25038312989/keyring/keychain.go#L44) to open the `.keychain-db` file instead. This works, but not consistently. `aws-vault add` now explodes.

To work around the issue, I ended up reverting that change, and symlinked `.keychain-db` to `.keychain`:

```
$ ls -l ~/Library/Keychains/aws-vault*
lrwxr-xr-x  1 dgoodlad  staff     21 20 Aug 14:07 /Users/dgoodlad/Library/Keychains/aws-vault.keychain -> aws-vault.keychain-db
-rw-r--r--  1 dgoodlad  staff  24668 20 Aug 14:50 /Users/dgoodlad/Library/Keychains/aws-vault.keychain-db
```

I'm not nearly familiar enough with what's going on here to work out the issue quickly, so I'm opening this issue to at least document what I know.

**Comment by lox** [comment]
Wow, that's terrible :( I wonder if they have exposed some API to deal with that. I'll do some digging.

**Comment by dgoodlad** [comment]
The really odd part is that the code for `aws-vault add` works despite using the old `.keychain` extension. My suspicion is there could be a way to continue referring to the keychain by its old name, but haven't tested it...

**Comment by dgoodlad** [comment]
I stand corrected on the odd behaviour of `aws-vault add` vs the other commands. I think this got mixed up as I was also debugging the crash in #83, and my brain cross-wired a few things 😄 

I've just tested with the simple change to open `~/Library/Keychains/aws-vault.keychain-db` instead of `.keychain`. I moved my keychain out of the way, and the symlink I'd made earlier, leaving nothing related to aws-vault in my Keychains directory. Everything works as it should.

I'll put together a quick PR to switch on OS version where we can discuss whether that's even a good idea 😉
